src/nitro/communication/messages/parser/nux/NewUserExperienceGiftOfferMessageParser.ts

```diff
@@ -0,0 +1,34 @@
+import { IMessageDataWrapper, IMessageParser } from '../../../../../api';
+import { NewUserExperienceGiftOptions } from '../../incoming/nux';
+
+export class NewUserExperienceGiftOfferMessageParser implements IMessageParser
+{
+    private _giftOptions: NewUserExperienceGiftOptions[];
+
+    public flush(): boolean
+    {
+        return true;
+    }
+
+    public parse(wrapper: IMessageDataWrapper): boolean
+    {
+        if(!wrapper) return false;
+
+        const _local_2 = wrapper.readInt();
+        this._giftOptions = [];
+        let _local_3:number;
```

Comment:
javascript doesnt initialize numbers by default to 0 like ActionScript. Initialize this to 0

Also rename it to `index`